### DISCUSSION TITLE
Does Zeek can be deployed on Android system ?

### DISCUSSION THREAD

**Discussion by cxdinter** [zeek/zeek#2366] [General]
Hi,
   I know Android compile strategy/environment is different with CMake. 
   Did someone tried to deploy Zeek on Android system? Is it possible ? 

   Thank you~

**Comment by ckreibich** [comment]
It's possible to build Zeek via Android's NDK, but that is only part of the puzzle — you'd at a minimum also need an Android-compatible libpcap to grab packets, and most likely a rooted phone to be able to do anything with the setup. If your main interest is to understand the traffic generated by specific Android apps, an easier route might be to set up a VPN on the device, routing traffic to an external machine that terminates the VPN and runs Zeek to inspect the traffic forwarded by the device.

**Reply by cxdinter** [reply]
Hi chreibich,
   Thanks for your quick response.
   I am working on Electronic vehicle field. Normally, the CDC(Cockpit Domain Controller) are using Qualcomm solution (SA8155/SA8295), and it is running Android system. I want to porting Zeek into CDC system to protect vehicle ethernet traffic. I have all necessary hardware and build environment. 
   Here, I just want to discuss the potential solution and efforts.

**Reply by ckreibich** [reply]
Cool! If you get it to work, you should definitely give a ZeekWeek talk about it. :-) We tinkered with running Zeek on Android in the context of past research projects, but mostly landed on the VPN approach. @narseo and @ssundaresan might recall details.

**Reply by narseo** [reply]
@ckreibich @cxdinter , we have been discussing different options to implement traffic filtering solutions on end-devices in the context of AppCensus, using either localhost or traditional VPN solutions. Happy to have a chat!
